crates/ironrdp-pdu/src/nego.rs

```diff
@@ -105,6 +105,11 @@ impl NegoRequestData {
     }
 
     pub fn read(src: &mut ReadCursor<'_>) -> PduResult<Option<Self>> {
+        // minimum length for token is 15
+        if src.len() < 15 {
+            return Ok(None);
+        }
+
```

Comment:
There is already a check here: https://github.com/Devolutions/IronRDP/blob/b98443b407187e9ab06bcdaf45270bd8a2443f75/crates/ironrdp-pdu/src/nego.rs#L372
I’m unsure how this is failing, but I would prefer we address it at this place instead of duplicating the check.

Reply:
This will return an Err NotEnoughBytes. We could eventually catch the error each time, but this is not much better.

Reply:
Good point! What about changing it to:
```rust
        // minimum length is length of prefix + crlf
        if src.len() < prefix.len() + 2 {
            return Ok(None);
        }
```

Reply:
hmm, I guess you mean min(Cookie::PREFIX.len(), RoutingToken::PREFIX.len())...

Reply:
At this place:
https://github.com/Devolutions/IronRDP/blob/b98443b407187e9ab06bcdaf45270bd8a2443f75/crates/ironrdp-pdu/src/nego.rs#L372
`prefix` will be either `Cookie::PREFIX` or `RoutingToken::PREFIX` depending on the caller:
https://github.com/Devolutions/IronRDP/blob/f209c9a77b834cf1f902084749e5c7316164a816/crates/ironrdp-pdu/src/nego.rs#L136
https://github.com/Devolutions/IronRDP/blob/f209c9a77b834cf1f902084749e5c7316164a816/crates/ironrdp-pdu/src/nego.rs#L155

Reply:
ok